tests/cnf/ran/ztp/tests/ztp-argocd-hub-templating.go

```diff
@@ -0,0 +1,218 @@
+		By("checking the ZTP version")
+		versionInRange, err := ranhelper.IsVersionStringInRange(RANConfig.ZTPVersion, "4.11", "")
+		Expect(err).ToNot(HaveOccurred(), "Failed to check if ZTP version is in range")
+
+		if !versionInRange {
+			Skip("This test requires a ZTP version of at least 4.11")
+		}
+
+		By("ensuring the test namespace exists")
+		_, err = namespace.NewBuilder(HubAPIClient, tsparams.TestNamespace).Create()
+		Expect(err).ToNot(HaveOccurred(), "Failed to create the test namespace")
+	})
+
+	AfterEach(func() {
+		By("resetting the policies app back to the original settings")
+		err := helper.SetGitDetailsInArgoCd(
+			tsparams.ArgoCdPoliciesAppName, tsparams.ArgoCdAppDetails[tsparams.ArgoCdPoliciesAppName], true, false)
+		Expect(err).ToNot(HaveOccurred(), "Failed to reset the git details for the policies app")
+
+		By("removing the hub templating leftovers if any exist")
+		network, err := sriov.PullNetwork(Spoke1APIClient, tsparams.TestNamespace, RANConfig.SriovOperatorNamespace)
+		if err == nil {
+			err = network.DeleteAndWait(tsparams.ArgoCdChangeTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to delete SR-IOV network")
+		}
+
+		By("removing the CGU if it exists")
+		cguBuilder, err := cgu.Pull(
+			HubAPIClient, tsparams.HubTemplatingCguName, tsparams.HubTemplatingCguNamespace)
+		if err == nil {
+			_, err = cguBuilder.DeleteAndWait(5 * time.Minute)
+			Expect(err).ToNot(HaveOccurred(), "Failed to delete and wait for CGU to be deleted")
+		}
+
+		By("removing test secret if it exists")
+		secretBuilder, err := secret.Pull(HubAPIClient, RANConfig.Spoke1Name+"-sriovdata", tsparams.TestNamespace)
+		if err == nil {
+			err = secretBuilder.Delete()
+			Expect(err).ToNot(HaveOccurred(), "Failed to delete test secret")
+		}
+	})
+
+	When("full access to ACM hub templating is used", func() {
+		BeforeEach(func() {
+			By("checking the TALM version")
+			versionInRange, err := ranhelper.IsVersionStringInRange(RANConfig.HubOperatorVersions[ranparam.TALM], "4.16", "")
+			Expect(err).ToNot(HaveOccurred(), "Failed to check if TALM version is in range")
+
+			if !versionInRange {
+				Skip("This test requires a TALM version of at least 4.16")
+			}
+		})
+
+		// 54240 - Hub-side ACM templating with TALM
+		It("should allow templating with the printf function", reportxml.ID("54240"), func() {
+			setupHubTemplateTest(tsparams.ZtpTestPathTemplatingPrintf)
+
+			By("validating the policy reaches compliant status")
+			policy, err := ocm.PullPolicy(HubAPIClient, tsparams.HubTemplatingPolicyName, tsparams.TestNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to get policy from hub cluster")
+
+			err = policy.WaitUntilComplianceState(policiesv1.Compliant, tsparams.ArgoCdChangeTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to wait for policy to become compliant")
+		})
+
+		// 54240 - Hub-side ACM templating with TALM
+		It("should allow templating with the prinf function", reportxml.ID("54240"), func() {
```

Comment:
description does not match test content. And it is now a valid test case. so should be moved to	`When("supported ACM hub side templating is used", func() {`